<a href="https://colab.research.google.com/github/RubenMtzE/Metodos_numericos/blob/main/Newton_Raphson_1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

<font size="5"> <div align="center">Métodos númericos</div></font> <br /><br />
<!-- \begin{center}
{\large Martínez Escutia Rubén}\\
Licenciatura en tecnología, Métodos númericos
\vspace{1.2cm}
\end{center} 
<br />
<br /> -->

La ecuación de estado de Beatle-Brifeman en su forma virial es:
$$PV=RT+\frac{\beta}{V}+\frac{\gamma}{V^2}+\frac{\delta}{V^3}$$\
Donde:  <br /><br />
P = Presión en atm \
T = Temperatura en Kelvin \
V = Volumen molar en $\frac{L}{g\cdot mol}$ \
R = Constante universal de los gases 0.08205 $\frac{atm\cdot L}{g\cdot mol\cdot K}$\
$\beta = RTB_0-A_0-\frac{Rc}{T^2}$\
$\gamma=-RTB_0b+A_0a-\frac{RB_0c}{T^2}$\
$\delta=\frac{RB_0bc}{T^2}$\
<!-- \begin{center}
\vspace{1cm}
\end{center} -->
Calcule el volumen molar V a 50 atm  y $100^{\circ}C$ para los siguientes gases 

| Gas | $A_0$ | a | $B_0$ | b | $c \times 10 ^{-4}$ |
| --- | --- | --- | --- | --- | --- |
| O$_2$| 1.4911 | 0.02562 | 0.04624 | 0.004208 | 4.8 | <br /><br /><br /><br />

Considerando la ecuación del gas ideal 

$$PV=nRT$$ \
Despejando V <br />
$$\frac{V}{n}=\frac{RT}{P}$$


In [ ]:
import pandas as pd

In [ ]:
#Definir el valor de las variables
P=50
T= 100 + 273.15
R= 0.08205
A0=1.4911
a=0.02562
B0=0.04624
b=0.004208
c= 4.8 *10**-4 
beta = R*T*B0-A0-((R*B0*c)/(T**2))
gamma = -R*T*B0*b+A0*a-((R*B0*c)/(T**2))
delta = (R*B0*b*c)/(T**2)
fx=[]
ii=[]

In [ ]:
#Evaluando la ecuación del gas ideal para conocer x0
V= (R * T )/ P
print(V)
#Evaluando beta, gamma, delta
print("beta = " , beta)
print("gamma = ",gamma)
print("delta = ",delta)

0.6123391499999999
beta =  -0.07537188521307893
gamma =  0.032244598079842715
delta =  5.503595253241735e-14


De la operación anterior consideraremos V como $x_0=0.6123...$

$$f(V)=RT+\frac{\beta}{V}+\frac{\gamma}{V^2}+\frac{\delta}{V^3}-PV$$
$$f(V)=RT+\beta V^{-1}+\gamma V^{-2}+\delta V^{-3}-PV$$
Derivando
$$f'(V)=-\frac{\beta}{V^2}-\frac{2\gamma}{V^3}-\frac{3\delta}{V^4}-P$$
Calculando $\Delta x$
$$\Delta x= \frac{f(V)}{f'(V)}=\frac{RT+\frac{\beta}{V}+\frac{\gamma}{V^2}+\frac{\delta}{V^3}-PV}{-\frac{\beta}{V^2}-\frac{2\gamma}{V^3}-\frac{3\delta}{V^4}-P}$$
Calculando $X_0$
$$X_0=X_0-\frac{f(V)}{f'(V)}$$
Para la primera iteracióón tenemos
$$X_0=0.6123-\frac{0.08205\times 373.15+\frac{-0.07537}{0.61233}+\frac{0.03224}{0.61233^2}+\frac{5.50359}{0.61233^3}-50\times0.61233}{-\frac{-0.07537}{0.61233^2}-\frac{0.03224}{0.61233^3}-\frac{5.50359}{0.61233^4}-50}=0.61159$$

In [ ]:
#Ajustar el valor en range() para el número de iteraciones.
for i in range(5):
  x = V
#Evaluar en la función f(x)
  gx= R*T+(beta/x)+(gamma/x**2)+(delta/x**3)-P*x
#Evaluar en la función f'(x)
  dgx=-(beta/x**2)-(((2*gamma))/x**3)-((3*delta)/x**4)-P
#Evaluar \Delta x
  Deltax = gx/dgx
#Evaluar xi
  V = x- Deltax
#Evaluando en la ecuación original
  gxn=R*T+(beta/V)+(gamma/V**2)+(delta/V**3)-P*V
  fx.append(V)
  ii.append(gxn)
  table={"f(x)":ii,"V":fx}
  print(V) 
tablesf = pd.DataFrame(table)
tablesf

0.6115984626169045
0.611598466565832
0.6115984665658322
0.611598466565832
0.6115984665658322


f(x)         V
0  1.977638e-07  0.611598
1  3.552714e-15  0.611598
2 -3.552714e-15  0.611598
3  3.552714e-15  0.611598
4 -3.552714e-15  0.611598

$\therefore$ el Volumen molar V a 50 atm y 100 $^{\circ}$C para el O$_2$es $$V=0.61159 \frac{L}{g\cdot mol}$$
En la primera iteración